backtester/engine/live_test.go

```diff
@@ -65,21 +64,33 @@ func TestSetupLiveDataHandler(t *testing.T) {
 
 func TestStart(t *testing.T) {
 	t.Parallel()
-	dc := &dataChecker{
-		shutdown: make(chan bool),
-	}
-	err := dc.Start()
-	assert.NoError(t, err)
+	t.Run("successful start", func(t *testing.T) {
+		t.Parallel()
+		dc := &dataChecker{
+			shutdown: make(chan bool),
+		}
+		err := dc.Start()
+		assert.NoError(t, err)
+
+		close(dc.shutdown)
+		dc.wg.Wait()
```

Comment:
🚧 I think this needs to use `assert.Eventually` 🙂 
It's much cleaner that way than letting the test just timeout blindly on failure, especially since you can have a comment to explain the expectation.

Shout if you need help with this.